### ISSUE TITLE
Nested Objects gives the incorrect result

### ISSUE THREAD

**Issue by daemonl** [jeremyfa/yaml.js#33]
Interesting issue in v0.2.1 (which is current, and the one on NPM - perhaps NPM should have the master/stable branch?)

``` yaml

---
key: { val1: "string", val2: { v2k1: "v2k1" } }
```

produces

``` json
{"key":{"val1":"string","val2":"string"}}
```

**Comment by jeremyfa** [comment]
Interesting indeed! well, several other issues were fixed when moving from 0.1.\* to 0.2.*, so I would rather investigate to see what is going on there instead of rollbacking the published version on NPM. Thank you for the feedback!

**Comment by jeremyfa** [comment]
It is now fixed in `v0.2.2` and a test has been added to be sure it won't break again.
